homeassistant/components/amcrest/__init__.py

```diff
@@ -222,68 +102,126 @@ def _start_event_monitor(hass, name, api, event_codes):
     thread.start()
 
 
-def setup(hass, config):
+async def async_setup(hass, config):
+    """Old way to set up Amcrest devices."""
+
+    # Import from yaml if exist
+    conf = config.get(DOMAIN)
+    if conf is not None:
+        for entry in conf:
+            # Only import if we haven't before.
+            config_entry = _async_find_matching_config_entry(hass)
+            if not config_entry:
+                if CONF_NAME not in entry:
+                    entry.update({CONF_NAME: DEFAULT_NAME})
+                if CONF_PORT not in entry:
+                    entry.update({CONF_PORT: DEFAULT_PORT})
+                hass.async_create_task(
+                    hass.config_entries.flow.async_init(
+                        DOMAIN, context={"source": SOURCE_IMPORT}, data=entry
+                    )
+                )
+
+    return True
+
+
+@callback
+def _async_find_matching_config_entry(hass):
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.source == config_entries.SOURCE_IMPORT:
+            return entry
+
+
+async def async_setup_entry(hass, config_entry):
     """Set up the Amcrest IP Camera component."""
     hass.data.setdefault(DATA_AMCREST, {DEVICES: {}, CAMERAS: []})
 
-    for device in config[DOMAIN]:
-        name = device[CONF_NAME]
-        username = device[CONF_USERNAME]
-        password = device[CONF_PASSWORD]
+    name = config_entry.data[CONF_NAME]
+    _LOGGER.debug("read config entry name : %s", name)
+    username = config_entry.data[CONF_USERNAME]
+    password = config_entry.data[CONF_PASSWORD]
+
+    api = AmcrestChecker(
+        hass,
+        name,
+        config_entry.data[CONF_HOST],
+        config_entry.data[CONF_PORT],
+        username,
+        password,
+    )
 
-        api = AmcrestChecker(
-            hass, name, device[CONF_HOST], device[CONF_PORT], username, password
-        )
+    ffmpeg_arguments = config_entry.options.get(
+        CONF_FFMPEG_ARGUMENTS, DEFAULT_FFMPEG_ARGUMENTS
+    )
+    resolution = config_entry.options.get(CONF_RESOLUTION, DEFAULT_RESOLUTION)
+    _LOGGER.debug("read config entry resolution : %s", resolution)
+    binary_sensors = config_entry.options.get(CONF_BINARY_SENSORS, [])
+    _LOGGER.debug("read binarey_sensor: %s", binary_sensors)
```

Comment:
Even if you do not consolidate the debug logging, there is a typo in the second one (binarey)